lib/active_merchant/billing/gateways/rapyd.rb

```diff
@@ -185,7 +187,9 @@ def add_ach(post, payment, options)
       end
 
       def add_token(post, payment, options)
-        post[:payment_method] = payment
+        return unless token = payment.split('|')[1]
+
+        post[:payment_method] = token
```

Comment:
If I'm understanding you correctly, we can't really test this remotely because of limitations with 3ds flow. But it seems like we should still have a unit test showing that a transaction with an authorization of the format `<auth_id>|<payment_method_id>` grabs the correct portion of the string and adds it to the request.

Maybe that seems unnecessary for such straightforward behavior, but it seems like the lack of test coverage is what let this slip through the cracks in the first place.

Reply:
I think this is a reasonable request. I'll add a specific check for this.